eth-types/src/sign_types.rs

```diff
@@ -0,0 +1,149 @@
+//! secp256k1 signature types and helper functions.
+
+use crate::{ToBigEndian, Word};
+use group::{ff::Field as GroupField, prime::PrimeCurveAffine, Curve, GroupEncoding};
+use halo2_proofs::arithmetic::{BaseExt, Coordinates, CurveAffine};
+use lazy_static::lazy_static;
+use num_bigint::BigUint;
+use secp256k1::Secp256k1Affine;
+use std::io::Cursor;
+use subtle::CtOption;
+
+/// Do a secp256k1 signature with a given randomness value.
+pub fn sign(
+    randomness: secp256k1::Fq,
```

Comment:
In general I think it would be best to pass an Rng just as you say; but in this particular case I wanted to sign with a specific randomness value; in particular for the default value of SignData, which must contain a verifiable signature, and I wanted the most simple values, so I chose `1` for the randomness: https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/25588c386e5a49d2ca8790c87b400a758b2e277c/eth-types/src/sign_types.rs#L57

Reply:
That makes sense then!